### ISSUE TITLE
Bare workflow app stuck on splash screen when expo-updates is enabled

### ISSUE THREAD

**Issue by 0duaht** [expo/expo#7896]
We made the switch to the bare workflow by ejecting out of the managed flow. While trying to setup OTA updates, we followed steps in the documentation - https://github.com/expo/expo/tree/master/packages/expo-updates, and have everything configured.

However, when we try running a release build for iOS, the app gets stuck on the splash screen each time EXUpdatesEnabled is turned on. The logs don't show any error

Things we've tried:
- Running `expo publish --release-channel channel` first to bundle and upload assets to Expo's host. Doing this fills in the right value for EXUpdatesURL using the exp.host URL for the project. However the app gets stuck on the Splash screen when run.
- Running `expo export` and then hosting the dist/ folder on a S3 bucket serving static files. Also gets stuck on the Splash screen.
- Setting `EXUpdatesEnabled` to false, app launches as usual. When set back to true, app gets stuck on the Splash screen.
- Cleaned build folder and tried a fresh build. Same error.

Expo version is 37.0.7, Expo-cli version is 3.17.24. Debug builds run fine.

**Comment by brentvatne** [comment]
hello! can you please provide a minimal reproducible example for us? there isn't much we can go on based on a report withot being able to reproduce it. more information on how to do that at https://stackoverflow.com/help/mcve

**Comment by 0duaht** [comment]
@brentvatne sure! I'm trying to reproduce with a fresh bare Expo project. Will update once I have more information

**Comment by edwardchen** [comment]
Getting something similar. Released SDK 37 on app last week without realizing OTA was enabled. Published expo today and now getting blank screen on my app in production. Error is: "RNCSafeAreaView" was not found in the UIManager.

I think it's due to upgrading react-navigation from v2 to v5. I think it might have to do with some missing linking that isn't handled in OTA (i think that's how it works?) Now I'm just waiting on apple to release my new version so my app stops crashing for people. (I guess I could publish an old version? seems like too much work)

Coming back to react native after a year and a half away from it and trying to upgrade everything has been quite the ordeal...

**Comment by edwardchen** [comment]
I published the old version and fixed the issue for now.

But if I publish a new version, as soon as anyone who has last week's version uses the new OTA update, it will break. So the risk for crashes will exist as long for anyone on this current app store release. I guess I can just avoid publishing for a while and hope people update the app.

**Comment by 0duaht** [comment]
Did some more digging. This happens when both `EXUpdatesRuntimeVersion` and `EXUpdatesSDKVersion` are present in the Expo.plist file.

Only setting `EXUpdatesSDKVersion` or `EXUpdatesRuntimeVersion` works smoothly.

The docs do say - `(exactly one of sdkVersion or runtimeVersion is required)`, but we took this to mean setting at least one of those values. Maybe rewording this to something like `(exactly one of sdkVersion or runtimeVersion should be set)`

To reproduce, run:
- `expo init -t bare-minimum sampleexpoupdatestest`
- Update `ios/sampleexpoupdatestest/Supporting/Expo.plist` to add a `EXUpdatesReleaseChannel` value
- Run `expo publish` | `expo export`. This automatically sets a value for `EXUpdatesSDKVersion` in Expo.plist as well
- Try creating and running a release build from Xcode, app stays stuck on the loading or splash screen.

**Comment by mdcuk34** [comment]
I also encountered this by misleadingly setting `EXUpdatesRuntimeVersion` and then running expo export which also set `EXUpdatesSDKVersion` causing the issue mentioned above.

**Comment by vittau** [comment]
I'm having this problem on Android as well, and I only have set `expo.modules.updates.EXPO_UPDATE_URL`, `expo.modules.updates.EXPO_SDK_VERSION` and `expo.modules.updates.EXPO_RELEASE_CHANNEL`.

I've followed [this](https://stackoverflow.com/a/61036446) to generate the build, and then `
bundletool build-apks --bundle=app-release.aab --output=app-release.apks --mode=universal` to generate an _APKS_ file, and then extracted `universal.apk` from it and used it on my real Android phone (Redmi Note 7).

**Comment by awinograd** [comment]
I'm similarly confused by the docs... It's not clear to me how to properly use EXUpdatesRuntimeVersion. There doesn't appear to be any way to set it with `expo publish` and I can't craft a curl against `exp.host` that returns a manifest if I dont send sdkVersion as a url param or header

**Comment by IsaacWIvins** [comment]
We had to use react-native-splash-screen in the mean time since this is broken. To be honest the expo-constants, expo-splash-screen, and expo-updates docs have been no help at all for the Bare Workflow

**Comment by brentvatne** [comment]
@IsaacWIvins sorry to hear that, this is the first release of expo updates and splash screen and we are working through some initial issues still. hopefully it’ll be sorted out for everyone soon.

**Comment by shubhamdeol** [comment]
Is this fixed?

**Comment by jmmunoz14** [comment]
not fixed yet, when i run the apk on adb with expo this issue happens.

**Comment by ex3ndr** [comment]
I hit pretty much the same issue, but have not found anything in sources about using and filtering updates via runtimeVersion. It seems that all work must be done via the backend (I am using a self-hosted one)

**Comment by brentvatne** [comment]
@ex3ndr - in self hosting you indeed need to do this entirely on your backend

**Comment by brentvatne** [comment]
> Did some more digging. This happens when both `EXUpdatesRuntimeVersion` and `EXUpdatesSDKVersion` are present in the Expo.plist file.
> 
> Only setting `EXUpdatesSDKVersion` or `EXUpdatesRuntimeVersion` works smoothly.
> 
> The docs do say - `(exactly one of sdkVersion or runtimeVersion is required)`, but we took this to mean setting at least one of those values. Maybe rewording this to something like `(exactly one of sdkVersion or runtimeVersion should be set)`
> 
> To reproduce, run:
> 
> * `expo init -t bare-minimum sampleexpoupdatestest`
> * Update `ios/sampleexpoupdatestest/Supporting/Expo.plist` to add a `EXUpdatesRuntimeVersion` value
> * Run `expo publish` | `expo export`. This automatically sets a value for `EXUpdatesSDKVersion` in Expo.plist as well
> * Try creating and running a release build from Xcode, app stays stuck on the loading or splash screen.

this looks like it has been resolved, i cannot repro it on the latest version of expo-updates: https://github.com/brentvatne/testing-double-field

if anyone encounters an issue like this again please post an issue and share a link to a minimal reproducible example repository along with instructions similar to the above, thanks!
